src/github.com/mozilla-services/pushgo/simplepush/structs.go

```diff
@@ -29,48 +28,32 @@ type PushCommand struct {
 }
 
 type PushWS struct {
-	uaidLock sync.RWMutex
-	uaid     string // Hex-encoded client ID; not normalized
-	Socket   Socket // Remote connection
+	uaid   string // Hex-encoded client ID; not normalized
+	Socket Socket // Remote connection
 	Store
 	Logger    *SimpleLogger
 	Metrics   *Metrics
 	Born      time.Time
-	closeLock sync.RWMutex
-	closed    bool
+	closeOnce Once
 }
 
 func (ws *PushWS) UAID() (uaid string) {
-	ws.uaidLock.RLock()
-	uaid = ws.uaid
-	ws.uaidLock.RUnlock()
-	return
+	return ws.uaid
 }
 
 func (ws *PushWS) SetUAID(uaid string) {
-	ws.uaidLock.Lock()
 	ws.uaid = uaid
-	ws.uaidLock.Unlock()
 }
 
 func (ws *PushWS) IsClosed() (closed bool) {
```

Comment:
From what I can see in the code, we're only calling this from Serv.Bye(). If we moved the Unregister and cleanup after the sock.Close() we might be able to remove that check as well.